pulsar-common/src/main/java/com/yahoo/pulsar/common/api/Commands.java

```diff
@@ -534,15 +534,19 @@ public static ByteBuf newConsumerStatsResponse(CommandConsumerStatsResponse.Buil
     private final static ByteBuf cmdPing = serializeWithSize(
             BaseCommand.newBuilder().setType(Type.PING).setPing(CommandPing.getDefaultInstance()));
 
-    static ByteBuf newPing() {
-        return RecyclableDuplicateByteBuf.create(cmdPing).retain();
+    public static ByteBuf newPing() {
+        ByteBuf pingCmdBuf =  RecyclableDuplicateByteBuf.create(cmdPing).retain();
```

Comment:
return `RecyclableDuplicateByteBuf.create(cmdPing)`

seems to work as well. With the old logic, it's ref count is initialized to 1 when we get it from recycler"RecyclableDuplicateByteBuf buf = RECYCLER.get();". The additional retain() is causing netty not to release it since it's ref count is not 0 after decrementing by 1?

Does it mean we have a leak? I don't see anything reported by leak detector though.